File: tests/Examples/openfhe/ext_const/ext_const_test.cpp
#include <cstdint>
#include <vector>

#include "gtest/gtest.h"  // from @googletest
#include "tests/Examples/openfhe/ext_const/ext_const_lib.h"

TEST(ExtConstTest, Basic) {
  std::vector<int32_t> res = test_fn();
  std::vector<int32_t> expected = {1, 2, 3, 4};
  EXPECT_EQ(res, expected);
}

File: tests/Examples/lattigo/bgv/ext_const/ext_const_test.go
package extconst

import (
	"testing"
)

func TestExtConst(t *testing.T) {
	res := Test_fn()
	t.Logf("Result: %v", res)
	if len(res) != 4 || res[0] != 1 || res[1] != 2 || res[2] != 3 || res[3] != 4 {
		t.Errorf("Expected [1 2 3 4], got %v", res)
	}
}
